File: packages/dotnet/src/endpoints/device_auth.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Json;
using System.Threading.Tasks;

namespace TavoAI
{
    /// <summary>
    /// Client for device_auth endpoints
    /// </summary>
    public class DeviceAuthClient
    {
        private readonly HttpClient httpClient;

        /// <summary>
        /// Create a new device_auth client
        /// </summary>
        /// <param name="httpClient">HTTP client instance</param>
        public DeviceAuthClient(HttpClient httpClient)
        {
            this.httpClient = httpClient;
        }

        /// <summary>
        /// POST /code
        /// </summary>
                /// <param name="client_id">client_id parameter</param>
        /// <param name="client_name">client_name parameter</param>
        public async Task<object> postcode(string? client_id = null, string? client_name = null)
        {
                    var url = "/code";
                    var content = JsonContent.Create(new { client_id = client_id, client_name = client_name });
                    var response = await this.httpClient.PostAsync(url, content);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// POST /token
        /// </summary>
                /// <param name="device_code">device_code parameter</param>
        public async Task<object> posttoken(string device_code)
        {
                    var url = "/token";
                    var content = JsonContent.Create(device_code);
                    var response = await this.httpClient.PostAsync(url, content);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// GET /info
        /// </summary>
                /// <param name="user_code">user_code parameter</param>
        public async Task<object> getinfo(string user_code)
        {
                    var queryParams = new Dictionary<string, object?> { "user_code", user_code };
                    var queryString = string.Join("&", queryParams.Where(p => p.Value != null).Select(p => $"{p.Key}={Uri.EscapeDataString(p.Value.ToString())}"));
                    var url = $"{formatted_path}" + (string.IsNullOrEmpty(queryString) ? "" : $"?{queryString}");
                    var content = null;
                    var response = await this.httpClient.GetAsync(url);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// POST /approve
        /// </summary>
        
        public async Task<object> postapprove()
        {
                    var url = "/approve";
                    var content = null;
                    var response = await this.httpClient.PostAsync(url, content);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// POST /code/cli
        /// </summary>
                /// <param name="client_name">client_name parameter</param>
        public async Task<object> postcodecli(string? client_name = null)
        {
                    var url = "/code/cli";
                    var content = JsonContent.Create(client_name);
                    var response = await this.httpClient.PostAsync(url, content);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// GET /code/{device_code}/status
        /// </summary>
                /// <param name="device_code">device_code parameter</param>
        public async Task<object> getcodestatus(string device_code)
        {
                    var queryParams = new Dictionary<string, object?> { "device_code", device_code };
                    var queryString = string.Join("&", queryParams.Where(p => p.Value != null).Select(p => $"{p.Key}={Uri.EscapeDataString(p.Value.ToString())}"));
                    var url = $"{formatted_path}" + (string.IsNullOrEmpty(queryString) ? "" : $"?{queryString}");
                    var content = null;
                    var response = await this.httpClient.GetAsync(url);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// GET /usage/warnings
        /// </summary>
        
        public async Task<object> getusagewarnings()
        {
                    var url = "/usage/warnings";
                    var content = null;
                    var response = await this.httpClient.GetAsync(url);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
        /// <summary>
        /// GET /limits
        /// </summary>
        
        public async Task<object> getlimits()
        {
                    var url = "/limits";
                    var content = null;
                    var response = await this.httpClient.GetAsync(url);
                    response.EnsureSuccessStatusCode();
                    return await response.Content.ReadFromJsonAsync<object>();
        }
    }
}

File: packages/typescript/src/endpoints/device_auth.ts
/**
 * Device_Auth API Client
 */

import { AxiosInstance } from 'axios'

export class Device_AuthClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * POST /code
   */
  async postCode(client_id?: string, client_name?: string): Promise<dict> {
        const params = {}
        const data = {'client_id: client_id, client_name: client_name'}
        return this.httpClient.post(`/code`, data, params)
  }
  /**
   * POST /token
   */
  async postToken(device_code: string): Promise<dict> {
        const params = {}
        const data = device_code
        return this.httpClient.post(`/token`, data, params)
  }
  /**
   * GET /info
   */
  async getInfo(user_code: string): Promise<dict> {
        const params = {'user_code: user_code'}
        const data = undefined
        return this.httpClient.get(`/info`, params)
  }
  /**
   * POST /approve
   */
  async postApprove(): Promise<dict> {
        const params = {}
        const data = undefined
        return this.httpClient.post(`/approve`, data, params)
  }
  /**
   * POST /code/cli
   */
  async postCodecli(client_name?: string): Promise<dict> {
        const params = {}
        const data = client_name
        return this.httpClient.post(`/code/cli`, data, params)
  }
  /**
   * GET /code/{device_code}/status
   */
  async getCode{device_code}status(device_code: string): Promise<dict> {
        const params = {'device_code: device_code'}
        const data = undefined
        return this.httpClient.get(`/code/{device_code}/status`, params)
  }
  /**
   * GET /usage/warnings
   */
  async getUsagewarnings(): Promise<dict> {
        const params = {}
        const data = undefined
        return this.httpClient.get(`/usage/warnings`, params)
  }
  /**
   * GET /limits
   */
  async getLimits(): Promise<dict> {
        const params = {}
        const data = undefined
        return this.httpClient.get(`/limits`, params)
  }
}
